File: 345-reverse-vowels-of-a-string/reverse-vowels-of-a-string.java
class Solution {
    public String reverseVowels(String s) 
    {
        String vowels = "aeiouAEIOU";

        char[] chars = s.toCharArray();

        int l = 0;
        int r = chars.length-1;

        while ( l < r)
        {
            while ( l < r && vowels.indexOf(chars[l]) == -1) l++;
            while ( l < r && vowels.indexOf(chars[r]) == -1) r--;

            char temp = chars[l];
            chars[l] = chars[r];
            chars[r] = temp;

            l++;
            r--;
        }
        
        return new String(chars);
    }
}

File: 345-reverse-vowels-of-a-string/reverse-vowels-of-a-string.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        vw = ['a','e','i','o','u','A','E','I','O','U']

        l = 0
        h = len(s)-1

        lst = list(s)

        while l < h:
            while l<h and s[l] not in vw:
                l += 1
            while l<h and s[h] not in vw:
                h -= 1
            lst[l] , lst[h] = lst[h] , lst[l]
            l += 1
            h -= 1

        return "".join(lst)
        